src/main/java/org/pentaho/di/trans/dataservice/serialization/DataServiceMetaStoreUtil.java

```diff
@@ -158,6 +159,15 @@ public DataServiceMeta getDataService( String serviceName, TransMeta serviceTran
         } catch ( Exception e ) {
           exceptionHandler.apply( e );
         }
+      } else {
+        try {
+          if ( PentahoSessionHolder.getSession() != null && SecurityHelper.getInstance().runAsSystem(
```

Comment:
@mkambol Moved check for session out of runAsSystem, this should prevent the NPE caused by service references missing in SecurityHelper when running on client.